src/bootstrap/builder.rs

```diff
@@ -1129,10 +1129,18 @@ impl<'a> Builder<'a> {
         // itself, we skip it by default since we know it's safe to do so in that case.
         // See https://github.com/rust-lang/rust/issues/79361 for more info on this flag.
         if target.contains("apple") {
-            if self.config.rust_run_dsymutil {
-                rustflags.arg("-Zrun-dsymutil=yes");
+            if stage == 0 {
+                if self.config.rust_run_dsymutil {
+                    rustflags.arg("-Zrun-dsymutil=yes");
+                } else {
+                    rustflags.arg("-Zrun-dsymutil=no");
+                }
             } else {
-                rustflags.arg("-Zrun-dsymutil=no");
+                if self.config.rust_run_dsymutil {
```

Comment:
Can this option be renamed? Also it would now be possible to ship rustc debuginfo as rustup component.

Reply:
My desired change of "I want my macOS build to be faster" has already snowballed into "I'm responsible for designing how to stabilize split-dwarf debuginfo" and a lot of other changes. I would rather not take on more responsibility in this PR for changing Rust's build system as well.